packages/core/src/render3/component_ref.ts

```diff
@@ -140,6 +140,9 @@ export class ComponentFactory<T> extends viewEngine_ComponentFactory<T> {
         rootViewInjector.get(RendererFactory2, domRendererFactory3) as RendererFactory3;
     const sanitizer = rootViewInjector.get(Sanitizer, null);
 
+    // Ensure that the namespace for the root node is correct,
+    // otherwise the browser might not render out the element properly.
+    ɵɵnamespaceHTML();
```

Comment:
Could we avoid calling instructions here?

Reply:
Do we have an alternative for it? I followed what we were doing inside `executeTemplate`.

Reply:
I think what he means is that we should create internal versions of the instructions rather than referencing them directly in non-generated code.

```suggestion
    namespaceHTMLInternal();
```